cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/batch/BatchConnector.java

```diff
@@ -35,10 +35,10 @@
   /**
    * Return the input format this connector will use to do the sampling
    */
-  InputFormatProvider getInputFormatProvider(ConnectRequest request);
+  InputFormatProvider getInputFormatProvider(SampleRequest request);
 
   /**
    * Transform the sampled records back to StructuredRecord
    */
-  List<StructuredRecord> transform(VAL_IN records);
+  List<StructuredRecord> transform(List<VAL_IN> records);
```

Comment:
Does this need to be List to List or can it just be VAL_IN to StructuredRecord? A List potentially has memory concerns.